lib/ContextExclusionPlugin.js

```diff
@@ -1,13 +1,28 @@
 "use strict";
 
+/** @typedef {import("./Compiler")} Compiler */
+/** @typedef {import("./ContextModuleFactory")} ContextModuleFactory */
+
 class ContextExclusionPlugin {
+	/**
+	 * @param {RegExp} negativeMatcher Matcher regular expression
+	 */
 	constructor(negativeMatcher) {
 		this.negativeMatcher = negativeMatcher;
 	}
 
+	/**
+	 * Apply the plugin
+	 * @param {Compiler} compiler Webpack Compiler
+	 * @returns {void}
+	 */
 	apply(compiler) {
-		compiler.hooks.contextModuleFactory.tap("ContextExclusionPlugin", cmf => {
-			cmf.hooks.contextModuleFiles.tap("ContextExclusionPlugin", files => {
+		compiler.hooks.contextModuleFactory.tap("ContextExclusionPlugin", (
+			/** @type {ContextModuleFactory} */ cmf
```

Comment:
I don't totally understand how tapable works but I think TypeScript can infer the types if `SyncHook` is somehow a generic of  `ContextExclusionPlugin` class. 

https://github.com/webpack/webpack/blob/4fc03e17d1a6f115b3b773ac0b663a73942d9d22/lib/Compiler.js#L44